### ISSUE TITLE
Is there any way to send cookies without using `request.setHeader` ?

### ISSUE THREAD

**Issue by reallyimeric** [arantes555/electron-fetch#3]
I can not send cookies via `{ session: session.fromPartition('myPartition') }` option, and it seems to be an issue of electron `net` module? https://github.com/electron/electron/issues/8891#issuecomment-286846734

I wonder is there a good way to send cookies (from session) currently?

**Comment by arantes555** [comment]
@reallyimeric Indeed, there is a problem with cookies, and it looks like electron has not yet fixed it. To my knowledge, the only way of handling cookies right now is not to use sessions but an external cookie jar, like ToughCookie, as I suggested in the comments of the issue you linked.

What you have to do is something like:
```javascript
import { Cookie, CookieJar } from 'tough-cookie'
let cookieJar = new CookieJar()

// ...

request.headers = {
  cookie: cookieJar
        .getCookiesSync(url, {allPaths: true})
        .map(cookie => cookie.cookieString())
        .join('; ')
}

// ...

for (const [name, value] of response.headers) {
  if (name === 'set-cookie') { // add new cookies to the cookie jar
    const cookie = Cookie.parse(value)
    cookieJar.setCookieSync(cookie, url)
  }
}
```

I'll leave this issue open until electron fixes net module, even if there really is nothing I can do.

**Comment by reallyimeric** [comment]
Thanks for your help.

**Comment by Tim91084** [comment]
@arantes555 from what I notice, the net module properly sets cookies in the session.  The issue is that (at least in the main process) the net module is not properly sending those cookies on each subsequent request.

Anyhow, I'm wondering what you think of this.. Before making a request, query the session for cookies using the function below:

```typescript
function getCookieHeader(url: string, electronSession: Electron.session): Promise<string> {

  return new Promise((resolve, reject) => {

    if (!electronSession) {
      reject('session is required')
      return
    }

    const cookies = electronSession.cookies.get({ url }, (error, sessionCookies) => {
      sessionCookies = sessionCookies || []
      const cookieHeader = sessionCookies.map(x => `${x.name}=${x.value}`).join(';')

      resolve(cookieHeader)
    })
  })
}
```

Then, add the cookie header to the request:

```typescript
...
  const cookieHeader = await getCookieHeader(url, options.session)

  if (cookieHeader) {
    options.headers[cookieHeaderName] = cookieHeader
  }

  return electronFetch(url, options)
```

**Comment by arantes555** [comment]
@Tim91084 That's a neat way of doing it, however, I'm not comfortable with integrating this into electron fetch directly, it feels to much of a hack to work-around a bug in electron.

**Comment by holmberd** [comment]
There seems like a fix when in for this in Electron 10: https://github.com/electron/electron/pull/22704/files

**Comment by arantes555** [comment]
@holmberd actually, this was also backported to all versions since 7.3, and it is already implemented in electron-fetch v1.6.0 (see https://github.com/arantes555/electron-fetch/blob/master/CHANGELOG.md#v160 ) :)
